hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/RootEntityResultImpl.java

```diff
@@ -105,6 +110,19 @@ public RootEntityResultImpl(
 			);
 		}
 
+		final EntityRowIdMapping rowIdMapping = entityDescriptor.getRowIdMapping();
+		if ( rowIdMapping == null ) {
+			rowIdResult = null;
+		}
+		else {
+			rowIdResult = rowIdMapping.createDomainResult(
+					navigablePath.append( rowIdMapping.getRowIdName() ),
+					entityTableGroup,
+					AbstractEntityPersister.ROWID_ALIAS,
+					creationState
+			);
+		}
+
```

Comment:
I think the same should be done for `org.hibernate.sql.results.graph.entity.internal.EntityResultImpl`

Reply:
I refactored the code and move the logic into `AbstractEntityResultGraphNode` instead. Thanks for the review.